From 2a06d75f1ff4fd357c2fa508fa054eb91c713c30 Mon Sep 17 00:00:00 2001
From: builder <robot@3mdeb.com>
Date: Tue, 17 Feb 2026 15:40:50 +0100
Subject: [PATCH] slaunch.c: Set MSI PRO B850 and MZ33-AR1 PSP version

MSI PRO B850 (1022:15C7) - change from PSP_NONE to PSP_V3
MZ33-AR1 (1022:156E) - add device and set to ASP (and expand
                       psp_version_t enum)

Upstream-Status: Pending
Signed-off-by: builder <robot@3mdeb.com>
---
 arch/x86/kernel/slaunch.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/slaunch.c b/arch/x86/kernel/slaunch.c
index 4274ada79274..157a494cea55 100644
--- a/arch/x86/kernel/slaunch.c
+++ b/arch/x86/kernel/slaunch.c
@@ -37,7 +37,8 @@ typedef enum {
 	PSP_NONE = 0,
 	PSP_V1,
 	PSP_V2,
-	PSP_V3
+	PSP_V3,
+	ASP
 } psp_version_t;
 
 static const struct pci_device_id psp_devs_list[] = {
@@ -48,7 +49,8 @@ static const struct pci_device_id psp_devs_list[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x15DF), .driver_data = PSP_V3 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x1649), .driver_data = PSP_V2 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x14CA), .driver_data = PSP_V3 },
-	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x15C7), .driver_data = PSP_NONE },
+	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x15C7), .driver_data =	PSP_V3 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, 0x156E), .driver_data =	ASP },
 	{ }
 };
 
-- 
2.39.2

